code/main/edge/edgeRequestWorker.brs

```diff
@@ -132,16 +133,58 @@ function _adb_EdgeRequestWorker() as object
             return responseArray
         end function,
 
-        _processRequest: function(eventData as object, ecid as string, configId as string, requestId as string, path as string, edgeDomain = invalid as dynamic) as object
-            requestBody = m._createEdgeRequestBody(eventData, ecid)
+        _processRequest: function(eventData as object, ecid as string, datastreamId as string, requestId as string, path as string, meta as object, edgeDomain as dynamic) as object
+            ''' Append config overrides to meta if set in eventData.config
+            meta = m._appendConfigOverridesToMeta(meta, eventData.config, datastreamId)
 
-            url = _adb_buildEdgeRequestURL(configId, requestId, path, edgeDomain)
+            ''' Override the datastreamId if set in eventData.config
+            datastreamId = m._getDatastreamId(eventData.config, datastreamId)
+
+            ''' Remove config from eventData
+            eventData.Delete("config")
+
+            requestBody = m._createEdgeRequestBody(eventData, ecid, meta)
+
+            url = _adb_buildEdgeRequestURL(datastreamId, requestId, path, edgeDomain)
             _adb_logVerbose("EdgeRequestWorker::_processRequest() - Processing Request with url:(" + chr(10) + FormatJson(url) + chr(10) + ") with payload:(" + chr(10) + FormatJson(requestBody) + chr(10) + ")")
             networkResponse = _adb_serviceProvider().networkService.syncPostRequest(url, requestBody)
             return networkResponse
         end function
 
-        _createEdgeRequestBody: function(eventData as object, ecid as string) as object
+        _getDatastreamId: function(config as object, originalDatastreamId as String) as string
+            if _adb_isEmptyOrInvalidMap(config)
+                return originalDatastreamId
+            end if
+
+            if not _adb_isEmptyOrInvalidString(config.datastreamIdOverride)
```

Comment:
I'm not 100% sure, but I guess `config.datastreamIdOverride` could cause errors if the key doesn't exist. Can we use `config["datastreamIdOverride"]` or call [Lookup(key as String) as Dynamic](https://developer.roku.com/docs/references/brightscript/interfaces/ifassociativearray.md#lookupkey-as-string-as-dynamic) function instead?

The same concern about the below code: 
```brightscript
            if not _adb_isEmptyOrInvalidMap(config.datastreamConfigOverride)
                meta["configOverrides"] = config.datastreamConfigOverride
            end if

            if not _adb_isEmptyOrInvalidString(config.datastreamIdOverride)
                ''' Genrate sdkConfig payload with original datastreamId
                meta["sdkConfig"] = m._getSdkConfigPayload(originalDatastreamId)
            end if
```

Reply:
I have added an test with config and it is working fine. So It shouldn't be a problem.